test/wallet-rpc-test.js

```diff
@@ -0,0 +1,338 @@
+/* eslint-env mocha */
+
+'use strict';
+
+const {NodeClient,WalletClient} = require('bclient');
+const assert = require('./util/assert');
+const FullNode = require('../lib/node/fullnode');
+const Network = require('../lib/protocol/network');
+const Mnemonic = require('../lib/hd/mnemonic');
+const HDPrivateKey = require('../lib/hd/private');
+const KeyRing = require('../lib/primitives/keyring');
+const Script = require('../lib/script/script');
+const Address = require('../lib/primitives/address');
+const network = Network.get('regtest');
+
+// commonly used test mnemonic
+const phrase = [
+  'abandon', 'abandon', 'abandon', 'abandon',
+  'abandon', 'abandon', 'abandon', 'abandon',
+  'abandon', 'abandon', 'abandon', 'about'
+].join(' ');
+
+const node = new FullNode({
```

Comment:
This test conflicts with a regtest running node, that has been previously resolved, see https://github.com/bcoin-org/bcoin/issues/717

Reply:
See the related fixes here: https://github.com/bcoin-org/bcoin/pull/719/files

Reply:
Good call on this. Will follow the same pattern

Reply:
This sounds like the type of thing to include in a `test/README.md`. Something along the lines of:

If you are writing a test that needs to start a server, please use a custom port such that the ports will not collide with the standard bitcoin ports. See test/http.js for an example.

Reply:
And documenting which ports are used for the test run as mentioned at https://github.com/bcoin-org/bcoin/pull/719#discussion_r263791381 and other resources used.

Reply:
Pending PR here that adds a `test/README.md`
https://github.com/handshake-org/hsd/pull/115

It may also be useful to define a `common.ports` so that we can define it once and use them between all the files